### PR TITLE
improve oauth handling

### PR DESCRIPTION
**Pull request by aenario**

### CODE REVIEW

**Comment by aenario** [comment]
Also include fix for https://github.com/cozy/cozy-emails/issues/548

**Comment by frankrousseau** [comment]
Could you add comment to the `account2config.coffee` file.

File: server/imap/account2config.coffee [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+CLIENT_ID = '260645850650-2oeufakc8ddbrn8p4o58emsl7u0r0c8s' +
+            '.apps.googleusercontent.com'
+
+CLIENT_SECRET = '1gNUceDM59TjFAks58ftsniZ'
+
+getXoauth2Generator = (account) ->
+    unless xOAuthCache[account.id]
+        log.info "XOAUTH GENERATOR FOR #{account.label}"
+
+        timeout = 1000*account.oauthTimeout - Date.now() #timeout in ms
```
Comment (frankrousseau) [inline server/imap/account2config.coffee]: space around operators is better.

File: server/imap/account2config.coffee [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        timeout = Math.floor timeout/1000 # timeout in s
```
Comment (frankrousseau) [inline server/imap/account2config.coffee]: same here.

File: server/imap/account2config.coffee [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                oauthTimeout: generator.timeout
+            , (err) ->
+                log.info "UPDATED ACCOUNT OAUTH #{account.label}"
+                log.warn err if err
+
+        xOAuthCache[account.id] = generator
+
+    return xOAuthCache[account.id]
+
+module.exports.forceOauthRefresh = (account, callback) ->
```
Comment (frankrousseau) [inline server/imap/account2config.coffee]: In the guidelines we decided to put two blank lines between functions.

**Comment by aenario** [comment]
Ok, will do the correction later.

**Pull request merged**